packages/client/hooks/useAutoCheckIn.ts

```diff
@@ -2,30 +2,48 @@ import graphql from 'babel-plugin-relay/macro'
 import {useEffect} from 'react'
 import {readInlineData} from 'relay-runtime'
 import {useAutoCheckIn_meeting} from '~/__generated__/useAutoCheckIn_meeting.graphql'
-import NewMeetingCheckInMutation from '../mutations/NewMeetingCheckInMutation'
+import JoinMeetingMutation from '../mutations/JoinMeetingMutation'
+import MeetingSubscription from '../subscriptions/MeetingSubscription'
 import useAtmosphere from './useAtmosphere'
+import useRouter from './useRouter'
 
 const useAutoCheckIn = (meetingRef: any) => {
   const atmosphere = useAtmosphere()
-  const {viewerId} = atmosphere
+  const {history, location} = useRouter()
+  const router = {history, location}
+  const queryKey = 'useAutoCheckIn'
   useEffect(() => {
     const meeting = readInlineData<useAutoCheckIn_meeting>(
       graphql`
         fragment useAutoCheckIn_meeting on NewMeeting @inline {
           id
           endedAt
           viewerMeetingMember {
-            isCheckedIn
+            id
           }
         }
       `,
       meetingRef
     )
     const {id: meetingId, endedAt, viewerMeetingMember} = meeting
-    if (endedAt) return
-    const {isCheckedIn} = viewerMeetingMember
-    if (!isCheckedIn) {
-      NewMeetingCheckInMutation(atmosphere, {meetingId, userId: viewerId, isCheckedIn: true})
+    const subscribeToMeeting = () => {
+      if (atmosphere.registerQuery) {
+        atmosphere.registerQuery(queryKey, MeetingSubscription, {meetingId}, router).catch()
+      }
+    }
+    if (viewerMeetingMember) {
+      subscribeToMeeting()
+    } else if (!endedAt) {
+      JoinMeetingMutation(
+        atmosphere,
+        {meetingId},
+        {onCompleted: subscribeToMeeting, onError: () => { }}
```

Comment:
if this is their first time in the meeting, subscribe to meeting events after they join